scripts/fastsp_script/fsp_test.py

```diff
@@ -1,114 +1,110 @@
 import os
 import sys
+import tempfile
 
 COM   = "," # uppercase is a visual indicator these are global constants
 SPACE = " " 
 SLASH = "/"
 FAS   = ".fasta"
+BAR   = "_"
 
 
 def parse_fastsp(if_lines, dataset, species, datatype, ref_num, est_num):
     b_SP = if_lines[0]
     b_TC = if_lines[5]
     
-    split_SP = b_SP.rsplit(space)
+    split_SP = b_SP.rsplit(SPACE)
     r_SP = split_SP[1]
     nn_SP = r_SP.rstrip()
-    split_TC = b_TC.rsplit(space)
+    split_TC = b_TC.rsplit(SPACE)
     r_TC = split_TC[1]
     #print(r_TC) 
     nn_TC = r_TC.rstrip()               
     
-    #write_line = file1 + com + file2 + com + file1_split[1]+ com +  nn_SP + com + nn_TC + "\n"
+    #write_line = file1 + COM + file2 + COM + file1_split[1]+ COM +  nn_SP + COM + nn_TC + "\n"
     # EMGT00050000000018.Drosophila.001_AA_13.fasta,EMGT00050000000018.Drosophila.001_AA_37.fasta,AA,0.9545768433975294,0.8126888217522659
-    write_line = dataset + com + species + com + datatype + com + ref_num + com + est_num + com + nn_SP + com + nn_TC + "\n"
+    global write_line
```

Comment:
**Never** define something as global, especially in a function!! Very very dangerous move that almost always leads to bugs, especially in this context when the function is being called over and over and over... What was the thought process of adding this line of code?

Reply:
See next comment.